### PR TITLE
feat: add "type" option to force interchange replacement method #10883

### PR DESCRIPTION
**Pull request by ncoden**
Add `type` option to Interchange:

> type {string} ['auto'] - Type of the responsive ressource to remplace. It can takes the following options:
> - `auto` (default): choose the type according to the element tag or the ressource extention
> - `src`: replace the `[src]` attribute, recommanded for images ``.
> - `background`: replace the `background-image` CSS property.
> - `html`: replace the element content.

Others changes:
* Clean some parts using ES6 features in Interchange

Closes https://github.com/zurb/foundation-sites/issues/10883

### CODE REVIEW

File: js/foundation.interchange.js [hunk 1]
```diff
@@ -83,6 +84,22 @@ class Interchange extends Plugin {
     }
   }
 
+  /**
+   * Check options valifity and set defaults for:
+   * - `data-interchange-type`: if set, enforce the type of remplacement (auto, src, background or html)
```
Comment (DanielRuf) [inline js/foundation.interchange.js]: Souldn't it be "replacement"?

File: js/foundation.interchange.js [hunk 2]
```diff
@@ -146,29 +163,39 @@ class Interchange extends Plugin {
...
 
-    var _this = this,
-        trigger = 'replaced.zf.interchange';
+    var trigger = 'replaced.zf.interchange';
+
+    var type = this.options.type;
+    if (type === 'auto') {
+      if (this.$element[0].nodeName === 'IMG')
+        type = 'src';
+      else if (path.match(/\.(gif|jpg|jpeg|png|svg|tiff)([?#].*)?/i))
```
Comment (DanielRuf) [inline js/foundation.interchange.js]: `jpe?g` could be also used. But what about webp and other formats?

File: js/foundation.interchange.js [hunk 3]
```diff
@@ -198,7 +225,19 @@ Interchange.defaults = {
    * @type {?array}
    * @default null
    */
-  rules: null
+  rules: null,
+
+  /**
+   * Type of the responsive ressource to remplace. It can takes the following options:
```
Comment (DanielRuf) [inline js/foundation.interchange.js]: remplace => replace
can takes => can take

File: js/foundation.interchange.js [hunk 4]
```diff
@@ -198,7 +225,19 @@ Interchange.defaults = {
...
+   * - `auto` (default): choose the type according to the element tag or the ressource extention
```
Comment (DanielRuf) [inline js/foundation.interchange.js]: extention => extension

File: js/foundation.interchange.js [hunk 5]
```diff
@@ -198,7 +225,19 @@ Interchange.defaults = {
...
+   * - `src`: replace the `[src]` attribute, recommanded for images `<img>`.
```
Comment (DanielRuf) [inline js/foundation.interchange.js]: recommanded => recommended

File: test/javascript/components/interchange.js [hunk 6]
```diff
@@ -95,6 +98,16 @@ describe('Interchange', function() {
       spy.restore();
     });
 
+    it('replaces following the `type` option', function() {
```
Comment (DanielRuf) [inline test/javascript/components/interchange.js]: replaces following the `type` option => replaces the following `type` option

**Review by DanielRuf — COMMENTED** [review]
Looks good so far but needs some changes in the texts.

File: js/foundation.interchange.js (hunk 1 above)
Comment (ncoden) [reply js/foundation.interchange.js]: It should. `remplacement` is french ;)

File: js/foundation.interchange.js (hunk 2 above)
Comment (ncoden) [reply js/foundation.interchange.js]: I kept the same regex as before. But you're right, any supported `` format should be added here.

**Comment by ncoden** [comment]
Thanks for the review. I shoud definitely pay more attention to typo.

**Pull request merged**